credential/src/test/java/io/syndesis/credential/CredentialsTest.java

```diff
@@ -264,4 +267,29 @@ public void shouldFinishOAuth2Acquisition() {
         assertThat(finalFlowState)
             .isEqualTo(new OAuth2CredentialFlowState.Builder().createFrom(flowState).accessGrant(accessGrant).build());
     }
+    
+    @Test()
+    public void checkTwitterOAuthException() {
+        String consumerKey="NMqaca1bzXsOcZhP2XlwA";
+        String consumerSecret="VxNQiRLwwKVD0K9mmfxlTTbVdgRpriORypnUbHhxeQw";
+        final OAuth1ConnectionFactory<?> oauth1 = new TwitterConnectionFactory(consumerKey, consumerSecret);
+        
+        String accessToken="26693234-W0YjxL9cMJrC0VZZ4xdgFMymxIQ10LeL1K8YlbBY";
+        String accessTokenSecret="BZD51BgzbOdFstWZYsqB5p5dbuuDV12vrOdatzhY4E";
+        properties.setAppId(accessToken);
+        properties.setAppSecret(accessTokenSecret);
+        
+        final Applicator<OAuthToken> applicator = new OAuth1Applicator(properties);
+        when(locator.providerWithId("providerId"))
+            .thenReturn(new OAuth1CredentialProvider<>("providerId", oauth1, applicator));
+        try {
+            @SuppressWarnings("unused")
+            final AcquisitionFlow acquisition = credentials.acquire("providerId", URI.create("https://syndesis.io/api/v1/"),
+                URI.create("https://syndesis.io/ui#state"));
+            Assert.fail();
+        } catch (HttpClientErrorException exception) {
+            //The HttpClientErrorExceptionMapper will pull out the message from here.
+            Assert.assertTrue(exception.getResponseBodyAsString().contains("<error>"));
```

Comment:
Perhaps use AssertJ for asserting exception is thrown?

https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion